src/cryptographic_primitives/secret_sharing/feldman_vss.rs

```diff
@@ -61,16 +61,15 @@ where
     // takes given VSS and generates a new VSS for the same secret and a secret shares vector to match the new coomitments
     pub fn reshare(&self) -> (VerifiableSS<P>, Vec<P::Scalar>) {
         let one: P::Scalar = ECScalar::from(&BigInt::one());
-        let poly = VerifiableSS::<P>::sample_polynomial(self.parameters.threshold.clone(), &one);
-        let index_vec: Vec<usize> = (1..=self.parameters.share_count.clone()).collect();
+        let poly = VerifiableSS::<P>::sample_polynomial(self.parameters.threshold, &one);
+        let index_vec: Vec<usize> = (1..=self.parameters.share_count).collect();
         let secret_shares_biased = VerifiableSS::<P>::evaluate_polynomial(&poly, &index_vec);
         let secret_shares: Vec<_> = (0..secret_shares_biased.len())
             .map(|i| secret_shares_biased[i].sub(&one.get_element()))
             .collect();
         let G: P = ECPoint::generator();
-        let mut new_commitments = Vec::new();
-        new_commitments.push(self.commitments[0].clone());
-        for i in 1..poly.len() {
+        let mut new_commitments = vec![self.commitments[0].clone()];
+        for (i, _) in poly.iter().enumerate().skip(1) {
```

Comment:
Now it looks more confusing to me than it was. I suggest to keep iterating over indexes (via `1..poly.len()`) or evolve suggested idea and use complex iterator:
```rust
for (i, (poly_i, commitment_i)) in poly.iter().zip(&self.commitments).enumerate().skip(1) {
    new_commitments.push((G.clone() * poly_i.clone()) + commitments_i.clone())
}
```

Reply:
Cool, blindly following clippy again :(

I played with this a bit and went with
```
let mut new_commitments = vec![self.commitments[0].clone()];
for (poly, commitment) in poly.iter().zip(&self.commitments).skip(1) {
    new_commitments.push((G.clone() * poly.clone()) + commitment.clone())
}
```

Do we want to assert that the arrays are the same length or is that guaranteed here already?

Reply:
> I played with this a bit and went with

Yes, you're right, your variant is better, we do not need index `i` anymore

> Do we want to assert that the arrays are the same length or is that guaranteed here already?

It is a good question. Let's assume here that it's guaranteed. Although VerifiableSS can be improperly constructed, we don't want this to cause a panic.